src/lib.rs

```diff
@@ -27,75 +27,96 @@
+//!     enable this fallback by adding the following to your crate's
+//!     `[dependancies]` section:
+//!     ```toml
+//!     getrandom = { version = "0.2", features = ["custom"]
+//!     ```
+//!
+//! - Use a custom `getrandom` implementation via an external crate.
+//!   - Some external creates define `getrandom` implementations for specific
+//!     unsupported targets.
+//!   - To use one of these crates (for example [`stdweb-getradnom`][23]):
+//!     - Add `stdweb-getrandom = "0.1"` to your crate's `[dependancies]` section.
+//!     - Use `stdweb-getrandom` in your crate. This can be done by adding
+//!       `use stdweb_getrandom as _;` to `src/lib.rs`. Explict usage is needed
+//!       to avoid linker errors.
+//!     - Using `getrandom` on `wasm32-unknown-unknown` will now call into the
+//!       implementation defined by the `stdweb-getrandom` crate.
+//!   - See [`register_custom_getrandom!`] for information about writing your
+//!     own custom `getrandom` implementation for an unsupported target.
+//!   - Custom implementations take precedence over the `"cpu"` feature.
+//!
+//! Both of these mechanisms only affect unsupported targets. Supported targets
+//! will always use their standard implementations.
+//!
+//! Of course, Pull Requests are welcome to add new targets to `getrandom`!
 //!
 //! ## Support for WebAssembly and asm.js
 //!
-//! Getrandom supports all of Rust's current `wasm32` targets, and it works with
-//! both Node.js and web browsers. The three Emscripten targets
-//! `asmjs-unknown-emscripten`, `wasm32-unknown-emscripten`, and
-//! `wasm32-experimental-emscripten` use Emscripten's `/dev/random` emulation.
-//! The WASI target `wasm32-wasi` uses the [`__wasi_random_get`][17] function
-//! defined by the WASI standard.
-//!
-//! Getrandom also supports `wasm32-unknown-unknown` by directly calling
-//! JavaScript methods. Rust currently has two ways to do this: [bindgen] and
-//! [stdweb]. Getrandom supports using either one by enabling the
-//! `wasm-bindgen` or `stdweb` crate features. Note that if both features are
-//! enabled, `wasm-bindgen` will be used. If neither feature is enabled, calls
-//! to `getrandom` will always fail at runtime.
-//!
-//! [bindgen]: https://github.com/rust-lang/rust-bindgen
+//! This crate supports the following `wasm32`/`asmjs` targets:
+//! - `wasm32-wasi`
+//! - `asmjs-unknown-emscripten`
+//! - `wasm32-unknown-emscripten`
+//!
+//! For `wasm32-unknown-unknown`, the story ismessy. This target can interface
+//! with JavaScript via either [wasm-bindgen] or [stdweb]. For this reason,
+//! **`wasm32-unknown-unknown` is an unsupported target**.
+//!
+//! However, as with any unsupported target, users can use custom `getrandom`
+//! implementation crates. For `wasm32-unknown-unknown`, there are:
+//! - [`wasm-bindgen-getrandom`][22], for use with [wasm-bindgen]
+//! - [`stdweb-getradnom`][23], for use with [stdweb]
```

Comment:
Better something like this IMO:

> The target `wasm32-unknown-unknown` is not directly supported since, from the target name alone, we cannot deduce which JavaScript interface is in use. Instead, the following crates provide support via the "custom `getrandom` implementation" mechanism (see above):
>
> - [`wasm-bindgen-getrandom`][22] via the [wasm-bindgen] JavaScript bridge
> ...

Reply:
Updated